### PR TITLE
Nginx server validation

### PR DESCRIPTION
**Pull request by dansterenson**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/47645

https://github.com/demisto/content/pull/18458

## Description
Added a validation that checks that the nginx server is up and running properly. In the future we plan to add safeguards to the [Safe indicator service](https://github.com/demisto/etc/issues/46470)

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [x] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/ApiModules/Scripts/NGINXApiModule/NGINXApiModule.py [hunk 1]
```diff
@@ -208,18 +208,22 @@ def nginx_log_monitor_loop(nginx_process: subprocess.Popen):
...
 
 
+def test_nginx_web_server(port: int, params: Dict):
+    protocol = 'https' if params.get('key') else 'http'
+    res = requests.get(f'{protocol}://localhost:{port}/nginx-test',
+                       verify=False, proxies={"http": "", "https": ""})  # nosec guardrails-disable-line
+    res.raise_for_status()
+    welcome = 'Welcome to nginx'
+    if welcome not in res.text:
+        raise ValueError(f'Unexpected response from nginx-text (does not contain "{welcome}"): {res.text}')
```
Comment (glicht) [inline Packs/ApiModules/Scripts/NGINXApiModule/NGINXApiModule.py:218]:
```suggestion
        raise ValueError(f'Unexpected response from nginx-test (does not contain "{welcome}"): {res.text}')
```

File: Packs/ApiModules/Scripts/NGINXApiModule/NGINXApiModule.py [hunk 2]
```diff
@@ -208,18 +208,22 @@ def nginx_log_monitor_loop(nginx_process: subprocess.Popen):
         nginx_log_process(nginx_process)
 
 
+def test_nginx_web_server(port: int, params: Dict):
+    protocol = 'https' if params.get('key') else 'http'
```
Comment (glicht) [inline Packs/ApiModules/Scripts/NGINXApiModule/NGINXApiModule.py:212]: Let's wrap the code block in a try/except and then re-raise the exception with a message that states `Failed testing nginx server: {ex}`

File: Packs/ApiModules/Scripts/NGINXApiModule/NGINXApiModule.py [hunk 3]
```diff
@@ -299,6 +303,9 @@ def run_long_running(params: Dict = None, is_test: bool = False):
 
         else:
             nginx_process = start_nginx_server(nginx_port, params)
+            # let nginx startup
+            time.sleep(0.5)
```
Comment (glicht) [inline Packs/ApiModules/Scripts/NGINXApiModule/NGINXApiModule.py:307]: Let's move this into the `test_nginx_web_server` server and also add a polling loop. So the function `test_nginx_web_server` will try the http request 5 times max each with a 0.5 delay. And only raise the exception after the 5th time.

File: Packs/EDL/ReleaseNotes/3_0_5.md [hunk 4]
```diff
@@ -0,0 +1,4 @@
+
+#### Integrations
+##### Generic Export Indicators Service
+- Added a validation that checks that the nginx server is running properly.
```
Comment (glicht) [inline Packs/EDL/ReleaseNotes/3_0_5.md:4]:
```suggestion
- Added a validation that checks that the nginx server is running properly after startup.
```

Comment (dansterenson) [reply Packs/EDL/ReleaseNotes/3_0_5.md]: Done

File: Packs/ApiModules/Scripts/NGINXApiModule/NGINXApiModule.py (hunk 3 above)
Comment (dansterenson) [reply Packs/ApiModules/Scripts/NGINXApiModule/NGINXApiModule.py]: Done

File: Packs/ApiModules/Scripts/NGINXApiModule/NGINXApiModule.py (hunk 2 above)
Comment (dansterenson) [reply Packs/ApiModules/Scripts/NGINXApiModule/NGINXApiModule.py]: Done

File: Packs/ApiModules/Scripts/NGINXApiModule/NGINXApiModule.py [hunk 5]
```diff
@@ -208,18 +210,36 @@ def nginx_log_monitor_loop(nginx_process: subprocess.Popen):
...
+                time.sleep(0.5)
+                protocol = 'https' if params.get('key') else 'http'
+                res = requests.get(f'{protocol}://localhost:{port}/nginx-test',
+                                   verify=False, proxies={"http": "", "https": ""})  # nosec guardrails-disable-line
+                res.raise_for_status()
+                welcome = 'Welcome to nginx'
+                if welcome not in res.text:
+                    raise ValueError(f'Unexpected response from nginx-test (does not contain "{welcome}"): {res.text}')
+                is_test_done = True
+            except HTTPError:
```
Comment (glicht) [inline Packs/ApiModules/Scripts/NGINXApiModule/NGINXApiModule.py:229]: Catch any exception. Not just HTTPError

**Review by glicht — APPROVED** [review]
Small comment about catching any exception

**Comment by dansterenson** [comment]
@yaakovi Force merge is needed

**Pull request merged**